File: Indusmática/5to ciclo/Progra 2/PRO2_LAB7_23-1-yoshi/Alumno.cpp
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/cppFiles/class.cc to edit this template
 */

/* 
 * File:   Alumno.cpp
 * Author: YOSHI
 * 
 * Created on 10 de junio de 2025, 15:54
 */

#include <cstring>
using namespace std;
#include "Alumno.h"

Alumno::Alumno() {
    construir();
}

Alumno::Alumno(const Alumno& orig) {
}

Alumno::~Alumno() {
}

void Alumno::construir(){
    nombre = nullptr;
    codigo = 0;
}

void Alumno::setTotal(double total) {
    this->total = total;
}

double Alumno::getTotal() const {
    return total;
}

void Alumno::setEscala(int escala) {
    this->escala = escala;
}

int Alumno::getEscala() const {
    return escala;
}

void Alumno::setNombre(const char* cad) {
    if (nombre!=nullptr) delete nombre;
    nombre = new char[strlen(cad)+1]{};
    strcpy(nombre,cad);
}

void Alumno::getNombre(char* cad) const {
    if (nombre==nullptr) cad[0] = 0;
    else strcpy(cad,nombre);
}

void Alumno::setCodigo(int codigo) {
    this->codigo = codigo;
}

int Alumno::getCodigo() const {
    return codigo;
}

File: Indusmática/5to ciclo/Progra 2/Lab10/2024-2/LAB10_Kenyi_Sin_Terminar/src/lab10_kenyi/Alumno.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package lab10_kenyi;

import java.util.Scanner;

/**
 *
 * @author kenyi
 */
public abstract class Alumno extends Registro{
    private String nombre;
    private int codigo;

    @Override
    public void cargar(Scanner arch) {
        codigo = arch.nextInt();
        nombre = arch.next();
    }

    @Override
    public void imprimir() {
        System.out.printf("%-12d %-60s",codigo,nombre);
    }
    
    public abstract double calcularPrecio(Producto produc,double tipoCambio);
    public abstract String getPaisOrigen();
//    public abstract double calcularPrecio()();
    
    /**
     * @return the nombre
     */
    public String getNombre() {
        return nombre;
    }

    /**
     * @param nombre the nombre to set
     */
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    /**
     * @return the codigo
     */
    public int getCodigo() {
        return codigo;
    }

    /**
     * @param codigo the codigo to set
     */
    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }
    
    
}
